iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -157,6 +157,75 @@ LogicalResult verifyTensorToVectorsPassPipelineConfig(
   return success();
 }
 
+LogicalResult verifyDoubleTilingExpertPassPipelineConfig(
+    Operation *op, IREE::Codegen::LoweringConfigAttr loweringConfig,
+    IREE::Codegen::TranslationInfoAttr translationInfo,
+    ArrayRef<int64_t> workgroupSize) {
+  if (!workgroupSize.empty()) {
+    return op->emitOpError(
+        "expected workgroup size to be empty for CPU pipelines");
+  }
+
+  // Verify that the translation info is using the right pipeline.
+  auto pipeline =
+      IREE::Codegen::DispatchLoweringPassPipeline::CPUDoubleTilingExpert;
+  StringRef pipelineName = stringifyEnum(pipeline);
+  if (translationInfo.getDispatchLoweringPassPipeline() != pipeline) {
+    return op->emitOpError("expected pipeline in translation.info to be ")
+           << pipelineName;
+  }
+
+  // Verify that the workload per workgroup is set and is non-zero.
+  SmallVector<int64_t> workloadPerWorkgroup =
+      translationInfo.getWorkloadPerWorkgroupVals();
+  if (workloadPerWorkgroup.size() > kNumMaxParallelDims) {
+    return op->emitOpError("workload_per_wg size should be less than ")
+           << kNumMaxParallelDims;
+  }
+  if (isa<linalg::LinalgOp, IREE::LinalgExt::TiledOpInterface>(op)) {
+    SmallVector<unsigned> partitionedLoops = getPartitionedLoops(op);
+    if (workloadPerWorkgroup.size() != partitionedLoops.size()) {
+      return op->emitOpError("expected ")
+             << partitionedLoops.size()
+             << " entries for workload_per_wg, but got "
+             << workloadPerWorkgroup.size();
+    }
+  }
+  if (llvm::any_of(workloadPerWorkgroup,
+                   [](int64_t val) { return val == 0; })) {
+    return op->emitOpError("invalid to use 0 in workload_per_wg");
+  }
+
+  if (loweringConfig.getTileSizes().size() != 2) {
+    return op->emitOpError("expected two levels of tile sizes for ")
+           << pipelineName << ", got " << loweringConfig.getTileSizes().size();
+  }
+  SmallVector<int64_t> firstLevelTileSizes = loweringConfig.getTileSizeVals(
+      static_cast<unsigned>(TilingLevel::WorkGroupTiles));
+  if (!firstLevelTileSizes.empty()) {
+    return op->emitOpError(
+        "expeted first level of tiling sizes are empty, which is applied at "
+        "flow level");
+  }
+
+  // Verify that native vector size is either empty, or if set is same as the
+  // last level of tiling
+  SmallVector<int64_t> nativeVectorSize =
+      loweringConfig.getNativeVectorSizeVals();
+  if (!nativeVectorSize.empty()) {
+    if (nativeVectorSize != loweringConfig.getTileSizeVals(
+                                static_cast<unsigned>(TilingLevel::L1Tiles))) {
+      return op->emitOpError(
+          "native_vector_size must be same as the last level of tiling");
+    }
+  }
```

Comment:
@MaheshRavishankar Are we going to deprecate native_vector_size? It's actually not used. I don't really know if want this check, this is sort of copying from another verifier.

Reply:
I'd like to deprecate `native_vector_size`. Should I just do it now?

Reply:
If it is not used, do whatever. Ill clean it up